src/k8s/pkg/k8sd/pki/control_plane.go

```diff
@@ -118,6 +118,11 @@ func (c *ControlPlanePKI) CompleteCertificates() error {
 		}
 		c.CACert = cert
 		c.CAKey = key
+	} else {
+		certCheck := pkiutil.CertCheck{CN: "kubernetes-ca", AllowSelfSigned: true}
```

Comment:
Thanks for the suggestion but again I think this would complicate the implementation unnecessarily. We have just one concrete implementation and we don't need to stub it for testing purposes, in which case I don't see the point of using an interface and dependency injection.